documentation/src/main/asciidoc/reference/search-dsl-predicate.asciidoc

```diff
@@ -784,11 +784,78 @@ but can be <<search-dsl-predicate-common-boost,boosted>>,
+
+.Matching a document that matches all the multiple given predicates (~`AND` operator)
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=and]
+----
+<1> The hits *must* have a `title` field matching the text `robot`,
+independently of other clauses in the same boolean predicate.
+<2> The hits *must* have a `description` field matching the text `crime`,
+independently of other clauses in the same boolean predicate.
+<3> All returned hits will match *all* the clauses above:
+they will have a `title` field matching the text `robot`
+*and* they will have a `description` field matching the text `crime`.
+====
+
+[[search-dsl-predicate-and-other]]
+=== Options
+
+* The score of an `and` predicate is variable by default,
+but can be <<search-dsl-predicate-common-constantScore,made constant with `.constantScore()`>>.
+* The score of an `and` predicate can be <<search-dsl-predicate-common-boost,boosted>>
+with a call to `.boost(...)`.
+
+[[search-dsl-predicate-or]]
+// TODO YB: find references to _combining_queries and add references to _oring_queries
+== [[_oring_queries]] `or`: combine predicates as a disjunction
+
+The `or` predicate matches documents that match at least one inner predicates, called "clauses".
+If none of the clauses match, then the `or` predicate will not match.
+
+Matching `or` clauses are taken into account during <<search-dsl-predicate-common-score,score>> computation.
+
+.Matching a document that matches any of multiple given predicates (~`OR` operator)
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/predicate/PredicateDslIT.java[tags=or]
+----
+<1> The hits *should* have a `title` field matching the text `robot`,
+*or* they should match any other clause in the same boolean predicate.
+<2> The hits *should* have a `description` field matching the text `investigation`,
+*or* they should match any other clause in the same boolean predicate.
+<3> All returned hits will match *at least one* of the clauses above:
+they will have a `title` field matching the text `robot`
+*or* they will have a `description` field matching the text `investigation`.
+====
+
+[[search-dsl-predicate-or-other]]
+=== Options
+
+* The score of an `or` predicate is variable by default,
+but can be <<search-dsl-predicate-common-constantScore,made constant with `.constantScore()`>>.
+* The score of an `or` predicate can be <<search-dsl-predicate-common-boost,boosted>>
+with a call to `.boost(...)`.
+
 [[search-dsl-predicate-boolean]]
-== [[_combining_queries]] `bool`: combine predicates (or/and/...)
+== [[_combining_queries]] `bool`: advanced combinations of predicates (or/and/...)
+
+The `bool` predicate allows to combine inner predicates in a more complex fashion than with simpler
```

Comment:
Grammar: it's either "allow <somebody> to -" or "allow -ing".
"allow to -" is incorrect.

```suggestion
The `bool` predicate allows combining inner predicates in a more complex fashion than with simpler
```

Reply:
Arf!  TIL that an infinitive complement shall only be used when a "noun phrase object (indicating the patient)" is specified, see e.g. https://english.stackexchange.com/questions/60271/grammatical-complements-for-allow/60285#60285.
Thanks for that lesson!

Reply:
I just edited my comment, because `<somebody>` was being interpreted as HTML and removed, which made my comment completely nonsensical :)